ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/QueryCompactor.java

```diff
@@ -54,6 +54,8 @@ public abstract class QueryCompactor implements Compactor {
   private static final Logger LOG = LoggerFactory.getLogger(QueryCompactor.class.getName());
   private static final String COMPACTOR_PREFIX = "compactor.";
 
+  protected CompactionQueryBuilderFactory compactionQueryBuilderFactory = new CompactionQueryBuilderFactory();
```

Comment:
Why is this ``protected``?

Reply:
Oh, sorry, forgot to send the answer here. I created an instance of the CompactionQueryBuilderFactory here, but it is used in the subclasses, that's why it is protected. But I realized, it is not a very good approach, so I changed it to create the instance where it is used.